allauth/socialaccount/providers/oauth2/views.py

```diff
@@ -16,22 +16,20 @@
 )
 from allauth.socialaccount.models import SocialLogin, SocialToken
 from allauth.socialaccount.providers.base import ProviderException
-from allauth.socialaccount.providers.oauth2.client import (
-    OAuth2Client,
-    OAuth2Error,
-)
+from allauth.socialaccount.providers.oauth2.client import OAuth2Client, OAuth2Error
 from allauth.utils import build_absolute_uri, get_request_param
 
 from ..base import AuthAction, AuthError
 
 
 class OAuth2Adapter(object):
-    expires_in_key = 'expires_in'
+    client_cls = OAuth2Client
```

Comment:
@benhowes Excellent idea! 💯🚀 I have also thought about it, but somehow did not take the courage to do this!